x-pack/plugins/ingest_manager/server/services/agent_policy.ts

```diff
@@ -388,9 +402,15 @@ class AgentPolicyService {
       throw new Error('Default output is not setup');
     }
     const defaultOutput = await outputService.get(soClient, defaultOutputId);
+    const settings = await getSettings(soClient);
```

Comment:
If settings does not exist, should we let `getSettings` throw a 404, or is it acceptable that the settings SO does not exist and we should still be able to return the full agent policy without an error?

Reply:
Good catch. This code is used to show the yaml config and this could happen before setting fleet, so we should probably swallow the 404 here.

Reply:
@nchaulet Thinking a bit more about this. Considering that just above, if no default output is setup, an error is thrown, I imagine it should be similar for Settings.  In this scenario that that setup hasn't run neither would exist and the output error would throw first, so perhaps we should just let settings 404, or catch it and throw a new error.

Reply:
Yes we can have a valid config without Kibana url in a case of a standalone config (an agent config to use without Fleet at all)

Reply:
Thanks @nchaulet.  I've updated so that it only adds when not in standalone and will throw an error if no settings are found in standalone.